pageserver/src/timelines.rs

```diff
@@ -114,8 +114,8 @@ impl LocalTimelineInfo {
 #[serde_as]
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct RemoteTimelineInfo {
-    #[serde_as(as = "Option<DisplayFromStr>")]
-    pub remote_consistent_lsn: Option<Lsn>,
+    #[serde_as(as = "DisplayFromStr")]
+    pub remote_consistent_lsn: Lsn,
```

Comment:
Before, we could have had a situation where `S3 list` shows that there's a remote timeline (so we add an info about it into the remote index), but we did not download the archive data to extract the `Lsn` info out of it.

Now we don't use S3 list and get a list of timeline files by downloading and reading `IndexShard` data, which also gives us the `TimelineMetadata` (with that `Lsn`), so every remote index entry is now instantly populated with all metadata available.